Description: introduce missing kfree
 Error handling code following a kzalloc should free the allocated data.
Origin: upstream, http://open-mesh.net/changeset/1492
Author: Julia Lawall <julia@diku.dk>

---
diff --git a/send.c b/send.c
index d724798278d6c813f9fafa88b4fb18bd394c1edf..eb617508cca4c5e8481b8729f50f0b692dae2c71 100644
--- a/send.c
+++ b/send.c
@@ -363,8 +363,10 @@ void add_bcast_packet_to_list(unsigned char *packet_buff, int packet_len)
 		return;
 
 	forw_packet->packet_buff = kmalloc(packet_len, GFP_ATOMIC);
-	if (!forw_packet->packet_buff)
+	if (!forw_packet->packet_buff) {
+		kfree(forw_packet);
 		return;
+	}
 
 	forw_packet->packet_len = packet_len;
 	memcpy(forw_packet->packet_buff, packet_buff, forw_packet->packet_len);
